### PR TITLE
chore(java): Extract BaseFury interface

### PR DESCRIPTION
**Pull request by LiangliangSui**
Extract the BaseFury interface based on Fury and ThreadSafeFury, and extract the common methods in Fury and ThreadSafeFury into BaseFury.

### CODE REVIEW

File: java/fury-core/src/main/java/org/apache/fury/BaseFury.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  <T> T deserializeJavaObject(byte[] data, Class<T> cls);
+
+  /**
+   * Deserialize java object from binary by passing class info, serialization should use {@link
+   * #serializeJavaObject}.
+   */
+  <T> T deserializeJavaObject(MemoryBuffer buffer, Class<T> cls);
+
+  /** Returns classLoader of serializer. */
+  ClassLoader getClassLoader();
```
Comment (chaokunyang) [inline java/fury-core/src/main/java/org/apache/fury/BaseFury.java:116]: Could we remove this API, it's unstable yet

Comment (LiangliangSui) [reply java/fury-core/src/main/java/org/apache/fury/BaseFury.java:116]: Move these two APIs back to `Fury` and `ThreadSafeFury`?

Comment (LiangliangSui) [reply java/fury-core/src/main/java/org/apache/fury/BaseFury.java:116]: Maybe move `ClassLoader getClassLoader()` this API back to `Fury` and `ThreadSafeFury`.

Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/BaseFury.java:116]: Yep, let's do this

Comment (LiangliangSui) [reply java/fury-core/src/main/java/org/apache/fury/BaseFury.java]: Done

**Review by chaokunyang — APPROVED** [review]
LGTM

**Pull request merged**